crates/cargo-test-support/src/paths.rs

```diff
@@ -236,6 +249,37 @@ impl CargoPathExt for Path {
             });
         }
     }
+
+    #[track_caller]
+    fn assert_file_layout(&self, expected: impl snapbox::IntoData) {
+        self.assert_file_layout_with_ignored_paths(expected, &default_ignored_paths());
+    }
+
+    #[track_caller]
+    fn assert_file_layout_with_ignored_paths(
+        &self,
+        expected: impl snapbox::IntoData,
+        ignored_paths: &[PathBuf],
+    ) {
+        let assert = assert_e2e();
+        let actual = WalkDir::new(self)
```

Comment:
Can we enable `sort_by_file_name()`?  Having a more consistent layout can improve the quality of the snapshots and make them easier to browse